docs/posts/dbt-ibis/index.qmd

```diff
@@ -0,0 +1,93 @@
+    first_name,
+    last_name,
+    first_order,
+    most_recent_order,
+    number_of_orders
+FROM customer_orders_info
+```
+
+dbt will make sure that the resulting table will be created after `stg_orders` and `stg_customers`.
+
+At the end of 2022, dbt added support for [Python models](https://docs.getdbt.com/docs/build/python-models) on specific platforms. This can be useful for complex transformations such as applying a machine learning model and storing the results. However, it also requires that your Python code is run in a cloud data warehouse and often, that data is moved into a Python process, which can be slower than leveraging the power of modern SQL engines.
+
+
+# Why dbt and Ibis go great together
+[dbt-ibis](https://github.com/binste/dbt-ibis) offers a more lightweight and compatible alternative, which allows you to write dbt models using Ibis. dbt-ibis transparently converts your Ibis statements into SQL and then hands it over to dbt. Your database does not need to have Python support for this as everything is executed in the same process as dbt. Rewriting the above SQL model in Ibis we get:
+
+```python
+from dbt_ibis import depends_on, ref
+
+
+@depends_on(ref("stg_customers"), ref("stg_orders"))
+def model(customers, orders):
+    customer_orders = orders.group_by("customer_id").aggregate(
+        first_order=orders["order_date"].min(),
+        most_recent_order=orders["order_date"].max(),
+        number_of_orders=orders.count(),
+    )
+    # Add first_name and last_name
+    customer_orders = customers.join(customer_orders, "customer_id", how="left")
+    return customer_orders.select(
+            "customer_id",
+            "first_name",
+            "last_name",
+            "first_order",
+            "most_recent_order",
+            "number_of_orders",
+        )
+```
+
+Using Ibis instead of SQL for dbt models brings you many advantages:
+
+* Type checks and validation before your code is even executed in a database
+* More composable as you can break down complex queries into smaller pieces
+* Better reusability of code. Although dbt allows you to use [Jinja and macros](https://docs.getdbt.com/docs/build/jinja-macros), which is an improvement over plain SQL, this gets you only so far. String manipulation is inherently fragile. With dbt-ibis, you can easily share common code between models.
+* Your dbt models become backend agnostic which reduces lock-in to a specific database. Furthermore, you get the possibility of building a [multi-engine data stack](https://juhache.substack.com/p/n-engines-1-language?utm_source=post-email-title&publication_id=1211981&post_id=137718100). For example, you could use DuckDB for small to medium workloads and Snowflake for heavy workloads and as an end-user and BI layer leveraging its governance features. Depending on the size of your warehouse, this can result in significant cost savings.
+* Unit test your code with your favorite Python testing frameworks such as pytest.
```

Comment:
this is pretty cool! they're working on adding unit testing to dbt now, but it will take a while -- this could be positioned as a good alternative in the meantime

Reply:
Yeah I saw it on the new roadmap, got me excited but I was also sad that a proper plugin system is not on the horizon for the next few months... Fully agree that this is a great alternative in the meantime :)